core/src/main/java/nl/nn/adapterframework/util/DB2XMLWriter.java

```diff
@@ -81,67 +89,169 @@ public static String getFieldType (int type) {
+			return xmlWriter.toString();
+		} catch (SAXException e) {
+			log.warn("cannot convert CallableStatement to XML", e);
+			return buildErrorString(e);
+		}
+	}
+
+	private static String buildErrorString(Exception e) {
+		return "<error>" + XmlEncodingUtils.encodeCharsAndReplaceNonValidXmlCharacters(e.getMessage()) + "</error>";
+	}
+
+	private static void addErrorXml(Exception e, SaxElementBuilder parent) throws SAXException {
+		SaxElementBuilder errorElement = parent.startElement("error");
+		errorElement.addValue(XmlEncodingUtils.encodeCharsAndReplaceNonValidXmlCharacters(e.getMessage()));
+		errorElement.endElement();
+	}
+
+	public void getXML(@Nonnull IDbmsSupport dbmsSupport, @Nonnull CallableStatement callableStatement, boolean alsoGetResultSets, @Nonnull Map<Integer, Parameter> outputParameters, int maxRows, boolean includeFieldDefinition, @Nonnull ContentHandler handler, boolean prettyPrint) throws SAXException {
+		// If a negative value is passed, retrieve all rows of each result set
+		if (maxRows < 0) {
+			maxRows = Integer.MAX_VALUE;
+		}
+		try (SaxDocumentBuilder root = new SaxDocumentBuilder("resultset", handler, prettyPrint)) {
+			try {
+				JdbcUtil.warningsToXml(callableStatement.getWarnings(), root);
+			} catch (SQLException e1) {
+				log.warn("exception obtaining statement warnings", e1);
+			}
+
+			if (alsoGetResultSets) {
+				processStatementResults(dbmsSupport, callableStatement, maxRows, includeFieldDefinition, root);
+			}
+
+			processOutputParameters(dbmsSupport, callableStatement, outputParameters, maxRows, includeFieldDefinition, root);
+		}
+	}
+
+	private void processOutputParameters(@Nonnull IDbmsSupport dbmsSupport, @Nonnull CallableStatement callableStatement, @Nonnull Map<Integer, Parameter> outputParameters, int maxRows, boolean includeFieldDefinition, @Nonnull SaxElementBuilder parent) throws SAXException {
+		if (outputParameters.isEmpty()) {
+			return;
+		}
+		StoredProcedureResultWrapper resultWrapper;
+		ResultSetMetaData metaData;
+		try {
+			resultWrapper = new StoredProcedureResultWrapper(dbmsSupport, callableStatement, callableStatement.getParameterMetaData(), outputParameters);
+			metaData = resultWrapper.getMetaData();
+		} catch (SQLException e) {
+			log.warn("Error get stored procedure result data", e);
+			addErrorXml(e, parent);
+			return;
+		}
+		int index = 1;
+		for (Map.Entry<Integer, Parameter> entry : outputParameters.entrySet()) {
+			SaxElementBuilder resultElement = parent.startElement(docname);
+			int position = entry.getKey();
+			Parameter param = entry.getValue();
+			resultElement.addAttribute("param", param.getName());
+			resultElement.addAttribute("type", param.getType().toString());
+
 			try {
-				ResultSetMetaData rsmeta = rs.getMetaData();
-				if (includeFieldDefinition) {
-					addFieldDefinitions(root, rsmeta);
+				callableStatement.getObject(position);
+				if (callableStatement.wasNull()) {
+					resultElement.addAttribute("null", "true");
```

Comment:
xsi gebruikt nil, misschien mooier dat aan te houden?
```suggestion
					resultElement.addAttribute("nil", "true");
```

Reply:
Dit komt overeen met wat de bestaande code doet in geval van `null` waarden, daarom denk ik dat het beter is om het attribuut `null` te laten heten.